packages/core-v2-sdk/src/datasources/TokenAPI/CoinGeckoAPIDataSource.ts

```diff
@@ -0,0 +1,19 @@
+import { HTTPDataSource } from "src/datasources/HTTPDataSource";
+
+export class CoinGeckoAPIDataSource extends HTTPDataSource {
+  static baseURL: "https://api.coingecko.com/api/v3/";
+
+  constructor() {
+    super("https://api.coingecko.com/api/v3/");
+  }
+
+  async getTokenPrice<ID extends string, CODE extends string>(
+    id: ID,
+    currency: CODE,
+  ): Promise<number> {
+    const res = await this.get<Record<ID, Record<CODE, number>>>(
+      `/simple/price?ids=${id}&vs_currencies=${currency ?? "usd"}`,
+    );
+    return res[id]?.[currency] ?? 1;
```

Comment:
we should let this function return null if a price could not be found instead of one